src/cli.js

```diff
@@ -177,6 +177,7 @@ async function bundle(main, command) {
   const Bundler = require('../');
 
   if (command.name() === 'build') {
+    command.watch = false;
```

Comment:
I think we should set `command.production = true` here actually, which would implicitly disable watching as well. I believe #1600 was meant to keep environment variables as set by the user, not control actual build. That can be done with other command line flags, e.g. if you want to disable minification, use `--no-minify`.